### PR TITLE
Add Link Sharing and Email Sharing

### PR DESCRIPTION
**Pull request by tikhsuP**
This PR adds the following to the share options:

 1. A link sharing option which, when clicked, copies the link of the archived page to the user's clipboard.
 2. A share through mail option.

### CODE REVIEW

**Comment by anishsarangi** [comment]
For email sharing, I could only see one option that is using `mailto: `, and you can try binding dynamically binding that through popup.js when the page loads.!

File: scripts/popup.js [hunk 1]
```diff
@@ -158,14 +165,31 @@ function social_share(eventObj) {
...
+          toastMsg[0].setAttribute('id', 'toast-success')
+          toastMsg[0].innerText = 'Page link copied.'
+          showToast(toastMsg[0])
+        }, function() {
+          toastMsg[0].setAttribute('id', 'toast-fail')
+          toastMsg[0].innerText = 'Error occurred. Try again.'
+          showToast(toastMsg[0])
+        })
+      } else if (id.includes('email')) {
+        $(document).ready(function() {
```
Comment (anishsarangi) [inline scripts/popup.js:180]: Why didi you add `$(document).ready()`? It will still work if you will not add that!!

File: scripts/popup.js [hunk 2]
```diff
@@ -498,6 +522,8 @@ $('#first_capture').click(first_capture)
 $('#fb_share').click(social_share)
 $('#twit_share').click(social_share)
 $('#linkedin_share').click(social_share)
+$('#copy-link').click(social_share)
+$('#email-share').click(social_share)
```
Comment (anishsarangi) [inline scripts/popup.js:526]: Add the href attribute directly inside `window.onload` function !

Comment (tikhsuP) [reply scripts/popup.js]: Thanks.

File: scripts/popup.js (hunk 1 above)
Comment (tikhsuP) [reply scripts/popup.js]: This function checks if the document is completely loaded or not. And performs the action only if it is needed. 

YES... there's no need to use this here. 
Thanks :)

File: scripts/popup.js (hunk 2 above)
Comment (tikhsuP) [reply scripts/popup.js]: I made the required changes. Is there a better way to do this?

File: scripts/popup.js [hunk 3]
```diff
@@ -484,7 +504,23 @@ chrome.runtime.onMessage.addListener(
...
+  var overview_url = 'https://web.archive.org/web/*/'
+  chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
+    let url = tabs[0].url
+    if (isNotExcludedUrl(url)) { // Prevents sharing some unnecessary page
+      if (url.includes('web.archive.org')) {
+        sharing_url = url // If the user is already at a playback page,share that URL
+      } else {
+        sharing_url = overview_url + get_clean_url(url) // When not on a playback page,share the overview version of that URL
+      }
+      $('#email-share').attr('href', `mailto:?subject=Hey, please check out this amazing website.&body=${sharing_url}`)
```
Comment (anishsarangi) [inline scripts/popup.js:518]: You can change the subject here like "Hey, Please checkout all archives of this website available in web.archive.org"

Comment (tikhsuP) [reply scripts/popup.js:518]: Sure, and, what about the "Toast" messages? Please tell me if you want any changes there :)

Comment (anishsarangi) [reply scripts/popup.js]: Looks fine to me !

**Comment by cgorringe** [comment]
Link Sharing got implemented in PR #683. We'll skip email sharing and close this now. Thanks!

**Pull request closed without merging**